src/connectionInfo/connectionManager.ts

```diff
@@ -0,0 +1,159 @@
+import { isIPv4 } from 'net';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import { DEFAULT_HOST } from '../utils/constants';
+import { Disposable } from '../utils/dispose';
+import { SETTINGS_SECTION_ID, SettingUtil } from '../utils/settingsUtil';
+import { Connection, ConnectionInfo } from './connection';
+
+/**
+ * @description keeps track of all of the Connection objects and the info needed to create them (ie: initial ports).
+ */
+const localize = nls.loadMessageBundle();
+export class ConnectionManager extends Disposable {
+	private _initHttpPort;
```

Comment:
I think it's a good habit to declare a type for all properties and variables that aren't immediately initialized (so the type can be inferred). It will make it easier to understand the intent and prevent mistakes